### PR TITLE
feat: for signup set make organization name optional and set a default one

### PR DESCRIPTION
**Pull request by zlwaterfield**
## Changes

This PR is making the organization name input for signup optional to reduce friction, it they don't enter one, it'll use `${first_name}'s Organization`. They can always update it later if needed. I also added an event to track how many people don't enter an organization name.

This PR also cleans up the first/last name usage to be more clear with consistent with other parts of the application.

## Does this work well for both Cloud and self-hosted?

it doesn't have an impact

## How did you test this code?

added e2e tests and unit tests

### CODE REVIEW

File: frontend/src/scenes/authentication/signup/signupForm/signupLogic.ts [hunk 1]
```diff
@@ -83,14 +84,20 @@ export const signupLogic = kea<signupLogicType>([
...
-            errors: ({ first_name, organization_name }) => ({
+            errors: ({ first_name }) => ({
                 first_name: !first_name ? 'Please enter your name' : undefined,
-                organization_name: !organization_name ? 'Please enter your organization name' : undefined,
             }),
             submit: async (payload, breakpoint) => {
                 breakpoint()
                 try {
-                    const res = await api.create('api/signup/', { ...values.signupPanel1, ...payload })
+                    const res = await api.create('api/signup/', {
```
Comment (xrdt) [inline frontend/src/scenes/authentication/signup/signupForm/signupLogic.ts:93]: Should we update the placdeholder for org to match the default: frontend/src/scenes/authentication/signup/signupForm/panels/SignupPanel2.tsx?

File: frontend/src/scenes/authentication/signup/signupForm/signupLogic.ts [hunk 2]
```diff
@@ -83,14 +84,20 @@ export const signupLogic = kea<signupLogicType>([
                 role_at_organization: '',
                 referral_source: '',
             } as SignupForm,
-            errors: ({ first_name, organization_name }) => ({
+            errors: ({ first_name }) => ({
                 first_name: !first_name ? 'Please enter your name' : undefined,
```
Comment (xrdt) [inline frontend/src/scenes/authentication/signup/signupForm/signupLogic.ts:88]: nit: we call this first_name but the form is labelled `Your name` so people probably end up entering first + last.

**Review by xrdt — APPROVED** [review]

File: frontend/src/scenes/authentication/signup/signupForm/signupLogic.ts (hunk 2 above)
Comment (zlwaterfield) [reply frontend/src/scenes/authentication/signup/signupForm/signupLogic.ts:88]: Agree, I will clean that up and make it more clear.

File: frontend/src/scenes/authentication/signup/signupForm/signupLogic.ts (hunk 1 above)
Comment (zlwaterfield) [reply frontend/src/scenes/authentication/signup/signupForm/signupLogic.ts:93]: Good call, yeah I can do that.

**Comment by zlwaterfield** [comment]
I've made some changes since this was approved, so wanted to get another set of eyes.

**Review by raquelmsmith — APPROVED** [review]
😎

**Pull request merged**